tests/lldp/test_lldp.py

```diff
@@ -33,11 +33,15 @@ def test_lldp(duthosts, enum_rand_one_per_hwsku_frontend_hostname, localhost,
         # Compare the LLDP neighbor name with minigraph neigbhor name (exclude the management port)
         assert v['chassis']['name'] == config_facts['DEVICE_NEIGHBOR'][k]['name']
         # Compare the LLDP neighbor interface with minigraph neigbhor interface (exclude the management port)
-        assert v['port']['ifname'] == config_facts['DEVICE_NEIGHBOR'][k]['port']
+        if request.config.getoption("--neighbor_type") == 'eos':
+            assert v['port']['ifname'] == config_facts['DEVICE_NEIGHBOR'][k]['port']
+        else:
+            # Dealing with KVM that advertises port description
+            assert v['port']['descr'] == config_facts['DEVICE_NEIGHBOR'][k]['port']
 
 
-def test_lldp_neighbor(duthosts, enum_rand_one_per_hwsku_frontend_hostname, localhost, eos,
-                       collect_techsupport_all_duts, loganalyzer, enum_frontend_asic_index, tbinfo):
+def test_lldp_neighbor(duthosts, enum_rand_one_per_hwsku_frontend_hostname, localhost, eos, sonic,
```

Comment:
Why do we need the additional parameter sonic?

Reply:
Both eos and sonic are needed to read and yield eos/sonic configuration.